zerver/tests/test_message_delete.py

```diff
@@ -224,6 +332,84 @@ def test_delete_message_by_other_user(msg_id: int) -> "TestHttpResponse":
         result = self.api_delete(test_bot, f"/api/v1/messages/{msg_id}")
         self.assert_json_success(result)
 
+    def test_delete_message_according_to_can_delete_any_message_group(self) -> None:
+        def check_delete_message_by_sender(sender_name: str, error_msg: str | None = None) -> None:
+            sender = self.example_user(sender_name)
+            msg_id = self.send_stream_message(sender, "Verona")
+            self.login_user(sender)
+            result = self.client_delete(f"/json/messages/{msg_id}")
+            if error_msg is None:
+                self.assert_json_success(result)
+            else:
+                self.assert_json_error(result, error_msg)
+
+        def check_delete_message_by_other_user(
+            sender_name: str, other_user_name: str, error_msg: str | None = None
+        ) -> None:
+            sender = self.example_user(sender_name)
+            other_user = self.example_user(other_user_name)
+            msg_id = self.send_stream_message(sender, "Verona")
+            self.login_user(other_user)
+            result = self.client_delete(f"/json/messages/{msg_id}")
```

Comment:
These might also be better written with `self.api_delete` to avoid changing the logged-in user, which is less composable.